dnfdaemon-server/services/goal/goal.cpp

```diff
@@ -74,6 +71,29 @@ sdbus::MethodReply Goal::resolve(sdbus::MethodCall & call) {
 
     auto reply = call.createReply();
     reply << result;
+
+    dnfdaemon::KeyValueMapList goal_resolve_log_list;
+
+    for (const auto & log : transaction.get_resolve_logs()) {
+        dnfdaemon::KeyValueMap goal_resolve_log_item;
+        goal_resolve_log_item["action"] = static_cast<uint32_t>(std::get<0>(log));
+        goal_resolve_log_item["problem"] = static_cast<uint32_t>(std::get<1>(log));
+        // TODO(nsella) better use of KeyValueMap with GoalJobSettings
+        dnfdaemon::KeyValueMap goal_job_settings;
+        goal_job_settings.emplace(std::make_pair("to_repo_ids", std::get<2>(log).to_repo_ids));
+        goal_resolve_log_item["goal_job_settings"] = goal_job_settings;
+        goal_resolve_log_item["report"] = std::get<3>(log); // string
+        goal_resolve_log_item["report_list"] = std::vector<std::string>{std::get<4>(log).begin(), std::get<4>(log).end()};
+        goal_resolve_log_list.push_back(goal_resolve_log_item);
+    }
+
+    dnfdaemon::KeyValueMap goal_resolve_results;
+    goal_resolve_results["transaction_problems"] = static_cast<uint32_t>(transaction.get_problems());
```

Comment:
We should try to find a better name for this than `transaction_problems`. It's more like exit code / status of the transaction resolving than problems. Those are returned in separate arrays.